src/main/java/org/opensearch/knn/common/KNNConstants.java

```diff
@@ -45,6 +45,7 @@ public class KNNConstants {
     public static final String MODEL_TIMESTAMP = "timestamp";
     public static final String MODEL_DESCRIPTION = "description";
     public static final String MODEL_ERROR = "error";
+    public static final String MODEL_NODE_ASSIGNMENT = "node_assignment";
```

Comment:
It's not inherently a duplicate, because they represent different things.  The KNNConstants class is used for all of the logic in the k-NN classes such as ModelDao, ModelMetadata and just the general code for that variable.  Meanwhile, the IndexUtil class is specifically a key for the versioning map.  In this situation, they happen to be the same but for consistency in both places I think we should keep both variables.

Reply:
I see. But this is used to determine what node the model is trained on correct? If so, they should have the same name. It should be "training_node_assignment". If it is just "node_assignment" it creates confusion around whether the model is cached on the node for use or if this is the node that the model was created on.

To avoid duplicate strings, you can set `MODEL_NODE_ASSIGNMENT_KEY = MODEL_NODE_ASSIGNMENT`

Reply:
Good call, will do that now